src/highlighters.cc

```diff
@@ -519,16 +520,26 @@ UniquePtr<Highlighter> create_line_highlighter(HighlighterParameters params, Hig
 }
 
 const HighlighterDesc column_desc = {
-    "Parameters: <value string> <face>\n"
-    "Highlight the column given by evaluating <value string> with <face>",
-    {}
+    "Parameters: [-ruler <character>] <column> <face>\n"
+    "Highlight the column <column> with <face>",
+    {
+        { { "ruler", { ArgCompleter{}, "replace empty or whitespace cells with the given character. When provided, <face> is not applied to non-empty cells" } } },
+        ParameterDesc::Flags::SwitchesOnlyAtStart, 2, 2
+    },
 };
 UniquePtr<Highlighter> create_column_highlighter(HighlighterParameters params, Highlighter*)
 {
-    if (params.size() != 2)
-        throw runtime_error("wrong parameter count");
+    ParametersParser parser{params, column_desc.params};
 
-    auto func = [col_expr=params[0], facespec=parse_face(params[1])]
+    auto positionals = parser.positionals_from(0);
+    auto ruler = parser.get_switch("ruler");
+    if (ruler.value_or(" ").char_length() > 1)
+        throw runtime_error("-ruler expects a single character");
+
+    auto ruler_provided = static_cast<bool>(ruler);
+    auto col_char = ruler.value_or(" ").str();
```

Comment:
Maybe do that before the char_length testing and test `col_char.char_length()` ?